### PR TITLE
Change default value for WC custom properties in servlet 5

### PR DESCRIPTION
**Pull request by pmd1nh**
Change the default value for Servlet 5 the following properties:

stopAppStartUponListenerException
decodeUrlPlusSign

fixes https://github.com/openliberty/open-liberty/issues/12020
fixes https://github.com/openliberty/open-liberty/issues/12290

### CODE REVIEW

**Comment by pmd1nh** [comment]
#build        #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Comment by pmd1nh** [comment]
need to update these tests for EE9
	
testDecodeUrlPlusSignDefault_EE9_FEATURES

File: dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -149,7 +149,7 @@ emptyServletMappings.name=Return empty servlet mappings
 emptyServletMappings.desc=Toggle if you want to return an empty collection, instead of null, when no servlet mappings are added. The default value is false. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.emptyservletmappings.
 
 deferServletRequestListenerDestroyOnError.name=Defer servlet request listener destroy
-deferServletRequestListenerDestroyOnError.desc=Toggle if you want to defer ServletRequestListener destroy when there is an error serving the request. The default value is false. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError.
+deferServletRequestListenerDestroyOnError.desc=Toggle if you want to defer ServletRequestListener destroy when there is an error serving the request. The default value is false. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError. For servlet-5.0 and newer Servlet features, the default value is true.
```
Comment (helyarp) [inline dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:152]: Either remove "The default value is false." or, preferably, change the description to:

Toggle if you want to defer ServletRequestListener destroy when there is an error serving the request. The equivalent custom property in the full application server profile is com.ibm.ws.webcontainer.deferServletRequestListenerDestroyOnError. For servlet-5.0 and newer Servlet features, the default value is true. For older Servlet features, the default is false.

Comment (pmd1nh) [reply dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:152]: @helyarp deferServletRequestListenerDestroyOnError is an old property (has been out there many years).  I don't know whether any customer may use any of these texts in their automation (they may not use or rely on the  description ...but I just don't know).

If it is necessary, I will make a change; but if you think it is ok, we should just leave it as-is.

**Comment by pmd1nh** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
Updated testDecodeUrlPlusSignDefault set of tests to run accordingly in EE9

**Comment by pmd1nh** [comment]
All WC passed, including the new added test for the servlet 5 decodeURLPLusSign default.


test_DecodeUrlPlusSignDefault_Servlet5_EE9_FEATURES | 1 | 0 | 0 | 0 s | 52
test_DecodeUrlPlusSignDefault_EE9_FEATURES | 1 | 0 | 0 | 0 s | 53
test_DecodeUrlPlusSign_EE9_FEATURES | 1 | 0 | 0 |  

2 failed tests are in the http2 and deployment (they are outside of these changes' scopes)
no component | 70410 | 2 | 0 |  
-- | -- | -- | -- | --

**Comment by pmd1nh** [comment]
@pnicolucci can you review it?

**Comment by pnicolucci** [comment]
@pmd1nh I'd like to see some tests delivered for `deferServletRequestListenerDestroyOnError` and `stopAppStartUponListenerException`

**Comment by pmd1nh** [comment]
added test for stopAppStartUponListenerException servlet5 default.

#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
added tests for stopAppStartUponListenerException pass

com.ibm.ws.fat.wc.tests.WCApplicationMBeanStatusTest | 8 | 0 | 0 | 2 s | 13
-- | -- | -- | -- | -- | --
test_AppStatusInstalled | 1 | 0 | 0 | 1 s | 1
test_AppStatusStarted | 1 | 0 | 0 | 0 s | 2
test_AppStatusInstalled_servlet5_default | 1 | 0 | 0 | 0 s | 3
test_AppStatusStarted_servlet5_default | 1 | 0 | 0 | 0 s | 4
test_AppStatusInstalled_EE9_FEATURES | 1 | 0 | 0 | 0 s | 5
test_AppStatusStarted_EE9_FEATURES | 1 | 0 | 0 | 0 s | 6
test_AppStatusInstalled_servlet5_default_EE9_FEATURES | 1 | 0 | 0 | 1 s | 7
test_AppStatusStarted_servlet5_default_EE9_FEATURES | 1 | 0 | 0 | 0 s |  

Other 3 failure are unrelated.

**Comment by pmd1nh** [comment]
#build        #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

**Comment by pmd1nh** [comment]
Tests: |   | 70523 tests, 0 failures, 0 errors
-- | -- | --

**Comment by pmd1nh** [comment]
#run-libby-bot

**Pull request merged**